modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeSetResultsSetTotalMethodCheck.java

```diff
@@ -5,79 +5,173 @@
+		matcher = _setTotalPattern.matcher(content);
 
-		return content;
+		String setTotalMethodCall = null;
+
+		int setTotalIndex = 0;
+
+		if (matcher.find()) {
+			setTotalIndex = matcher.start();
+
+			setTotalMethodCall = JavaSourceUtil.getMethodCall(
+				content, setTotalIndex);
+		}
+
+		if (!hasClassOrVariableName(
+				"SearchContainer", content, newContent, fileName,
+				setResultMethodCall) &&
+			!hasClassOrVariableName(
+				"SearchContainer", content, newContent, fileName,
+				setTotalMethodCall)) {
+
+			return newContent;
+		}
+
+		return StringUtil.replace(
+			newContent, content,
+			_getNewMethodCall(
+				content, setResultIndex, setResultMethodCall, setTotalIndex,
+				setTotalMethodCall));
 	}
 
-	private String _removeSetTotal(String content, String fileName) {
-		String newContent = content;
+	private String _formatJava(String content, String fileName)
+		throws Exception {
 
-		Matcher setTotalMatcher = _setTotalPattern.matcher(content);
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
 
-		while (setTotalMatcher.find()) {
-			if (hasClassOrVariableName(
-					"SearchContainer", content, content, fileName,
-					JavaSourceUtil.getMethodCall(
-						content, setTotalMatcher.start()))) {
+		if (javaClass == null) {
+			return content;
+		}
 
-				newContent = StringUtil.removeSubstring(
-					newContent, setTotalMatcher.group());
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
 			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			content = _formatContent(javaMethodContent, content, fileName);
 		}
 
-		return newContent;
+		return content;
 	}
 
-	private String _replaceSetResults(String content, String fileName) {
-		String newContent = content;
+	private String _getNewMethodCall(
+		String javaMethodContent, int setResultIndex,
+		String setResultMethodCall, int setTotalIndex,
+		String setTotalMethodCall) {
 
-		Matcher setResultsMatcher = _setResultsPattern.matcher(content);
+		String newJavaMethodContent = javaMethodContent;
 
-		while (setResultsMatcher.find()) {
-			String methodCall = JavaSourceUtil.getMethodCall(
-				content, setResultsMatcher.start());
+		List<String> parameters = new ArrayList<>();
 
-			if (hasClassOrVariableName(
-					"SearchContainer", content, content, fileName,
-					methodCall)) {
+		String methodCall = null;
 
-				newContent = StringUtil.replace(
-					newContent, methodCall,
-					StringUtil.replace(
-						methodCall, ".setResults(", ".setResultsAndTotal("));
+		if ((setResultMethodCall != null) && (setTotalMethodCall != null)) {
+			methodCall =
+				(setResultIndex > setTotalIndex) ? setResultMethodCall :
+					setTotalMethodCall;
+
+			String firstMethodCall = setResultMethodCall;
+
+			if (methodCall.contains("setResults")) {
+				firstMethodCall = setTotalMethodCall;
 			}
```

Comment:
@TamyrisTorres, aren't these 2 checks redundant? Don't you need only the variable `methodCall`?

How about you create a clone of the `methodCall` and use its clone in the call `StringUtil.removeString` in order to preserve the original `methodCall`?

Something like:
```
String methodCallCopy = methodCall;
```